server/query/cursor.cpp

```diff
@@ -153,7 +154,28 @@ std::tuple<Cursor::Process, Result> Cursor::ExecuteVelox(
   SDB_ASSERT(!batch);
   SDB_ASSERT(_runner);
   yaclib::Future<> wait;
-  auto r = basics::SafeCall([&] { batch = _runner.Next(wait); });
+
+  std::exception_ptr sql_error;
+  auto r = basics::SafeCall([&] {
+    try {
```

Comment:
avoid safecall, because now you have try catch
also my suggestion is to avoid catching wrapped exception if it's SqlException